provider/datagen/src/transform/cldr/plurals/mod.rs

```diff
@@ -48,37 +33,41 @@ impl PluralsProvider {
     }
 }
 
-impl<M: KeyedDataMarker<Yokeable = PluralRulesV1<'static>>> DataProvider<M> for PluralsProvider {
-    fn load(&self, req: DataRequest) -> Result<DataResponse<M>, DataError> {
-        Ok(DataResponse {
-            metadata: Default::default(),
-            payload: Some(DataPayload::from_owned(PluralRulesV1::from(
-                #[allow(clippy::unwrap_used)] // TODO(#1668) Clippy exceptions need docs or fixing.
-                self.get_rules_for(M::KEY)?
+macro_rules! implement {
+    ($marker:ident) => {
+        impl DataProvider<$marker> for crate::DatagenProvider {
+            fn load(&self, req: DataRequest) -> Result<DataResponse<$marker>, DataError> {
+                Ok(DataResponse {
+                    metadata: Default::default(),
+                    payload: Some(DataPayload::from_owned(PluralRulesV1::from(
+                        #[allow(clippy::unwrap_used)]
+                        // TODO(#1668) Clippy exceptions need docs or fixing.
+                        self.get_rules_for(<$marker>::KEY)?
+                            .0
+                            .get(&req.locale.get_langid())
+                            .ok_or(DataErrorKind::MissingLocale.into_error())?,
+                    ))),
```

Comment:
Nit (optional): There is not too much code here, so it's probably fine to leave it inside the macro, but you could still move this into a helper function like `fn get_payload_for` and then call it as `payload: Some(self.get_payload_for(<$marker>::KEY))`

(also in IterableDataProvider)